### PR TITLE
ISSUE #293 Enhancement: Add Dark Mode

### PR DESCRIPTION
**Pull request by zolomohan**
## Dark Mode

image

This can be attained by passing a prop to Picker Component.

`import { Picker } from 'emoji-mart'`
...
`<Picker darkMode={true}>`

The Default Prop is false.

This PR Fixes Issue #293 . Enhancement: Add Dark Mode

### CODE REVIEW

File: css/emoji-mart.css [hunk 1]
```diff
@@ -4,6 +4,11 @@
   line-height: 1.15;
 }
 
+.emoji-mart-dark-bg{
+  color: #fff !important;
+  background-color: #222 !important;
+}
```
Comment (nolanlawson) [inline css/emoji-mart.css]: Using `!important` here feels a bit hacky. Can we be more targeted with this?

File: css/emoji-mart.css [hunk 2]
```diff
@@ -151,13 +160,17 @@
...
   font-weight: 500;
   padding: 5px 6px;
-  background-color: #fff;
-  background-color: rgba(255, 255, 255, .95);
+  background-color: rgb(255, 255, 255, 0.95);
+  color: #000;
+}
+
+.emoji-mart-category-label-span-dark {
+  background-color: rgba(0, 0, 0, 0.95) !important;
```
Comment (nolanlawson) [inline css/emoji-mart.css]: Ditto about `!important` here.

File: src/utils/shared-default-props.js [hunk 3]
```diff
@@ -31,6 +31,7 @@ const PickerDefaultProps = {
   emojisToShowFilter: null,
   showPreview: true,
   showSkinTones: true,
+  darkMode: false,
```
Comment (nolanlawson) [inline src/utils/shared-default-props.js]: The default should probably be based on prefers-color-scheme: https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme

**Comment by nolanlawson** [comment]
I also think a simpler solution may be to just add the `emoji-mart-dark` CSS to the top level, and then use CSS to select for that (rather than add a new class to all the children). For instance:

```css
.emoji-mart-dark .something-inside-of-emoji-mart {
  background: black;
}
```

**Comment by nolanlawson** [comment]
I made the changes suggested in my PR review and pushed it above ^. I only tweaked the design slightly (changed the background color of the sections and added a white fill to the search icon):

Screenshot from 2019-12-21 09-09-39

**Comment by nolanlawson** [comment]
I'm going to merge as two separate PRs. Thanks very much for this!

**Comment by nolanlawson** [comment]
Closing in favor of https://github.com/missive/emoji-mart/pull/391

**Comment by zolomohan** [comment]
> I'm going to merge as two separate PRs. Thanks very much for this!

You're very welcome!

**Pull request closed without merging**